packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -555,9 +555,48 @@ export class StripeWebhookHandler extends StripeHandler {
 
   /**
    * Handle `invoice.paid` Stripe wehbook events.
+   *
+   * This handler sends out our invoice emails to customers as well as verifying
+   * that we are not accepting payments for users that may have been deleted or
+   * have a Stripe customer account with subscription that has become unlinked
+   * from a FxA account. We have observed this can happen in some edge cases during
+   * checkout in the past, and want to capture sufficient information for Support
+   * to manually handle.
    */
   async handleInvoicePaidEvent(request: AuthRequest, event: Stripe.Event) {
     const invoice = event.data.object as Stripe.Invoice;
+    const customer = await this.stripeHelper.expandResource(
+      invoice.customer,
+      CUSTOMER_RESOURCE
+    );
+    let invalidCustomer = false;
+    // Store as much relevant user data as we can lookup in case they were
+    // deleted or not linked to a FxA account.
+    const deletedData: Record<string, any> = {
+      customerId: invoice.customer,
+      invoiceId: invoice.id,
+    };
+    if (!customer || customer?.deleted) {
+      invalidCustomer = true;
```

Comment:
You know the use case for this better than I do, but if you think it might be helpful to know why should we add a `reason` property to the `deletedData` object? There seems to be 3 branches where `invalidCustomer` can be set to true: 1. `!customer || customer?.deleted` 2. `!uid` and 3. `!account` would it be helpful to either Support or for troubleshooting to know which branch was evaluated? 

We could add something like `deletedData.reason = 'no account'` or something to the respective branches. Not a blocker, especially if this isn't useful.

Reply:
Ah, I think this is a good idea, I'll add it.